test/unit/cmap/wire_protocol/compression.test.ts

```diff
@@ -0,0 +1,52 @@
+import { expect } from 'chai';
+
+import { compress, Compressor, decompress } from '../../../../src/cmap/wire_protocol/compression';
+
+describe('compression', function () {
```

Comment:
You mentioned wanting to add more integration-y tests, I think we could use them. 
Maybe you can use sinon spy to listen to push on `MessageStream`? and check that buffers begin with  `0xFD2FB528` which should be the [magic number](https://github.com/facebook/zstd/blob/dev/doc/zstd_compression_format.md#frames) for zstd. I think you can do the same to these unit tests instead of a not.equals check looking for the magic number seems like a good assertion.

Reply:
Unit tests are now checking for the magic number.